app/hotels/src/singleHotel/PaymentScreen.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import * as React from 'react';
+import { Platform } from 'react-native';
+import { WebView } from '@kiwicom/react-native-app-common';
+import moment from 'moment';
+
+export type PaymentParameters = {
+  hotelId: number,
+  checkin: Date,
+  checkout: Date,
+  rooms: Array<{|
+    id: string,
+    count: number, // how many rooms with this ID?
+  |}>,
+  affiliateId: string,
+  language: string,
+  currency: string,
+};
+
+export default function PaymentScreen(props: PaymentParameters) {
+  return <WebView source={{ uri: createURI(props) }} />;
+}
+
+export function createURI(pp: PaymentParameters) {
+  const checkinQuery = moment(pp.checkin).format('YYYY-MM-DD'); // Booking.com
+  const intervalQuery = moment(pp.checkout.getTime()).diff(
+    moment(pp.checkin.getTime()),
+    'days',
+  );
+
+  const roomsQuery = pp.rooms
+    .map(({ id, count }) => `nr_rooms_${id}=${count}`)
+    .join('&');
+
+  return (
+    'https://secure.booking.com/book.html' +
+    '?from_source=hotel' +
+    '&hostname=hotels.kiwi.com' +
```

Comment:
not so nice 🙂  I'm thinking whether something simple like ```Object.entries({ a: 'foo', b: 'bar'}).reduce((memo, [key, value]) => memo + `&${key}=${value}` , '')``` would help, not sure

Reply:
Rather something like this:

```js
querystring.stringify({
  from_source: 'hotel',
  ...
});
```

Reply:
Alright - it's "nicer" now... 🙂